src/blr/cli/filterclusters.py

```diff
@@ -54,9 +55,12 @@ def strip_barcode(pysam_read, tags_to_be_removed, removed_tags):
 
     # Remove tags
     for bam_tag in tags_to_be_removed:
-        removed_tags[bam_tag].add(pysam_read.get_tag(bam_tag))
-        # Strip read from tag
-        pysam_read.set_tag(bam_tag, None, value_type="Z")
+
+        if pysam_read.has_tag(bam_tag):
+            removed_tags[bam_tag].add(pysam_read.get_tag(bam_tag))
+            # Strip read from tag
+            pysam_read.set_tag(bam_tag, None, value_type="Z")
+            summary[f"Total {bam_tag} tags removed"] += 1
```

Comment:
If the previous comment is true, there is no need to pass summary to this function, it can be incremented outside.

Reply:
See my comment ;)